shlr/meson.build

```diff
@@ -1,211 +1,35 @@
-    capstone_version = 'v5'
-  endif
-
-  res = run_command(py3_exe, '-c', '__import__("sys").exit(__import__("os").path.exists("@0@"))'.format(capstone_path))
-  if res.returncode() == 0
-    if not git_exe.found()
-      error('Cannot load capstone library. Either provide capstone in ./shlr/capstone or install git, so it can be downloaded')
-    endif
-
-    # NOTE: when you update CS_TIP or CS_BRA, also update them in shlr/Makefile
-    if capstone_version == 'v5'
-      CS_TIP = 'd956ceddabcbe544ee8f074ba75c86c6381b34e6'
-      CS_BRA = 'next'
-    elif capstone_version == 'v3'
-      CS_TIP = '61bf71c771680033651f16cff832446e421847b1'
-      CS_BRA = 'v3'
-    elif capstone_version == 'v4'
-      CS_TIP = '2edae851d9fee511a57d4da32d5acecedd95d7ed'
-      CS_BRA = 'v4'
-    else
-      error('Wrong capstone version selected. Please use one of the supported versions.')
-    endif
-
-    capstone_git_user = 'aquynh'
-
-    message('Cloning capstone ' + CS_BRA + ' branch, commit ' + CS_TIP + ', into ' + capstone_path)
-    git_cmd = 'clone -b @0@ https://github.com/@1@/capstone.git @2@'.format(CS_BRA, capstone_git_user, capstone_path)
-    clone_cmd = run_command(git_exe, git_cmd.split())
-    if clone_cmd.returncode() != 0
-      error('Cannot execute git clone command')
+  if capstone_version == 'bundled'
+    r = run_command(py3_exe, check_meson_subproject_py, 'capstone-bundled')
+    if r.returncode() == 1 and get_option('subprojects_check')
+      error('Subproject capstone-bundled is not updated. Please make sure to run `meson subprojects update` or set option `subprojects_check=false` if you want to compile against currently downloaded subproject.')
     endif
-
-    reset_cmd = run_command(py3_exe, git_exe_repo_py, git_exe, capstone_path, 'reset', '--hard', CS_TIP)
-    if reset_cmd.returncode() != 0
-      error('Cannot execute git reset command')
+    capstone_proj = subproject('capstone-bundled', default_options: ['default_library=static'])
+  elif capstone_version == 'next'
+    r = run_command(py3_exe, check_meson_subproject_py, 'capstone-next')
+    if r.returncode() == 1 and get_option('subprojects_check')
+      error('Subproject capstone-next is not updated. Please make sure to run `meson subprojects update` or set option `subprojects_check=false` if you want to compile against currently downloaded subproject.')
     endif
-
-    patches_files = [
-      'capstone-calloc.patch',
-      'fix-x86-16.patch',
-      'sparc-crash.patch',
-      'sstream-null.patch'
-    ]
-
-    message('Patching capstone with rizin\'s patches')
-    foreach file : patches_files
-      patch_path = join_paths(meson.current_source_dir(), 'capstone-patches', file)
-      patch_cmd = run_command(py3_exe, git_exe_repo_py, git_exe, capstone_path, 'apply', '-p1', patch_path)
-      if patch_cmd.returncode() != 0
-        warning('Cannot apply patch ' + file)
-      endif
-    endforeach
-  endif
```

Comment:
We don't need to patch capstone anymore?

Reply:
See capstone bundled wrap. We have a fork there and i already applied the patches in the fork